scripts/github_api

```diff
@@ -0,0 +1,118 @@
+
+    try:
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as exc:
+        if r.status_code in [CLIENT_ERROR]:
+            error_msg = response['errors'][0]['message']
+            raise Exception(error_msg)
+        raise
+
+    return response
+
+
+class PullRequest:
+    """
+    https://developer.github.com/v3/pulls/
+    """
+    def __init__(self, repo):
+        self.repo = repo
+        self._endpoint = API_BASE_URL + "repos/%(repo)s/pulls"
+
+    def open(self, title, body, src_branch, dst_branch):
+        endpoint = self._endpoint % dict(repo=self.repo)
+        payload = {
+            "title": title,
+            "body": body,
+            "head": src_branch,
+            "base": dst_branch,
+        }
+
+        self._set_pr_info(request('POST', endpoint, data=json.dumps(payload)))
+
+    def _set_pr_info(self, pr_info):
+        for k, v in pr_info.items():
+            if type(v) != dict:
+                setattr(self, k, v)
+
+
+def open_pull_request(title, src_branch, repo, dst_branch, body=""):
+    pr = PullRequest(repo)
+    pr.open(title, body, src_branch, dst_branch)
+
+
+if __name__ == '__main__':
+    commands = {
+        'open_pr': {
+            'description': "Opens GitHub pull-requests",
+            'function': open_pull_request,
+            'arguments': [
+                (('title',), dict(help='pull request title text')),
+                (('src_branch',), dict(metavar='src-branch',
+                    help='pull request source branch: [user:]<branch>')),
+                (('repo',), dict(
+                    help='destination repository: <repo owner>/<repo>')),
+                (('dst_branch',), dict(metavar='dst-branch',
+                    help='pull request destination branch: <branch>')),
+                (('--body',), dict(help='pull request body text')),
+            ],
+        },
+    }
+
+    parser = argparse.ArgumentParser(description='Wrapper to GitHub API')
+    parser.add_argument('user', help='github user name')
+    parser.add_argument('password', help='github password or token')
+
+    subparsers = parser.add_subparsers(dest='command')
+
+    for cmd_name, cmd in commands.items():
+        _parser = subparsers.add_parser(cmd_name,
+                                        description=cmd['description'])
+
+        for arg_name, arg_opts in cmd['arguments']:
+            _parser.add_argument(*arg_name, **arg_opts)
+
+    args = parser.parse_args()
+    cmd_parser = subparsers.choices[args.command]
+
+    # just so we can use **cmd_args below. The slicing of argv is to
+    # exclude the positional params in the main parser plus the
+    # program and command names which are not known by the subparser
+    cmd_args = vars(cmd_parser.parse_args(sys.argv[4:]))
+
+    _setup_auth(args.user, args.password)
+
+    for key, value in cmd_args.items():
+        if value is None:
+            cmd_args.pop(key)
+
+    commands[args.command]['function'](**cmd_args)
```

Comment:
I'd prefer to split libraries and executables in separate files, as we probably will want to call this GitHub API from our code in the future, but up to you